lib/kubeclient/common.rb

```diff
@@ -477,5 +482,16 @@ def http_options(uri)
 
       options.merge(@socket_options)
     end
+
+    def check_token_expiration(http_code, message)
+      if http_code == 401 &&
+         @auth_options.key?(:bearer_token_expiry) &&
+         @auth_options[:bearer_token_expiry] < Time.now
+        return message + '. Your token has expired. If you are using `kubectl`, running any '\
+          'command should renew your token.'
```

Comment:
I opened #213 which provides support for using Google's default application credentials. I think this is a better solution as long as you are fine with a dependency on `googleauth` gem.